### DISCUSSION TITLE
[v4.1] Knex: Timeout acquiring a connection. The pool is probably full on createMany

### DISCUSSION THREAD

**Discussion by wemersonrv** [adonisjs/core#1993] [Help] [answered]
Hello, 

The Database has any limitations?

I am running a `createMany` with good amount of data... and receive some of these warnings: **Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?**

Sometimes, happens a timeout exception with the same message: *SubError [TimeoutError]: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?*

How can i fix it?

**Comment by thetutlage** [comment, marked as answer]
It is hard to know, what you mean by "good amount of data". Maybe share the number of rows you are trying to insert. Also, the number of connections are never infinite. If AdonisJS will not limit them, then you DB server will. 

Here's a guide on [connection pooling](https://preview.adonisjs.com/guides/database/connections-management#connection-pooling) and same applies on v4 as well

**Reply by wemersonrv** [reply]
I have 8 queue jobs with bull queues working in background.  They can works all in same time, or just a few of them. 7 works with only 1 record per time and it doesn't matter if is insert, update or delete; it is always one record at a time in transaction. and i have just one task that process huge more data per time... may vary from100 to 500 records in the same transaction... the media is 200. I think this is the issue!

Now I limited my loop here to 10 records on createMany and then commit the transaction. Does that work?

There have not yet been more than 200 records, only one with 124, so we had 13 createMany/commit without fail.
